cip-0005.md

```diff
@@ -0,0 +1,81 @@
+
+Being able to let all XCP owners vote on a particular issue can be very useful to poll the consensus / desire of the community.
+
+The votes done in this way is not part of this CIP and the initial intention is not to have any protocol changes activate automatically based on these votes,
+though that could be a future CIP if there's a desire to do so.
+
+As discussed in the community, to avoid making anyone have more power within the protocol, we won't be using tokens
+but will instead use a meta protocol with messages encoded in broadcast messages.
+
+This also makes this easily usable for other assets.
+
+We'll add the code to track votes to counterparty-lib/client so that anyone can easily track ongoing votes without requiring extra scripts.
+We'll also add code to counterwallet to easily cast votes, this should cover a big part of the XCP holders.
+
+## Specification ##
+
+There will be 2 types of 'special' broadcast messages that are used for this; `INITVOTE` and `CASTVOTE`.
+
+#### `INITVOTE`
+```INITVOTE <VOTENAME> <ASSET> <DEADLINE> (<STAKEHEIGHT>)? OPTS <OPTION_1> <OPTION_N>```
+ - `VOTENAME`: any sequence of non-space characters used as label for the vote, must be unique across the network.
+ - `ASSET`: the asset for which the olders can vote.
+ - `DEADLINE`: the deadline (block height or timestamp of block, see below) for the voting period (int).
+ - `STAKEHEIGHT`: the height at which to take the balances of `ASSET` to use to weight the votes.
+   if `STAKEHEIGHT` is ommitted then the block in which the `INITVOTE` broadcast is confirmed is used.
+ - `OPTS`, followed by 2+ options, ie; `OPTS TRUE FALSE`.
+   options are seperated by a space, any other characters are allowed by for simplicity only using alphanumeric chars is recommended.
+
+##### `DEADLINE`
+`DEADLINE` &lt; 500000000 Block number at which this poll ends, &gt;= 500000000	UNIX timestamp at which this poll ends
+(same as bitcoin nlocktime).
+A maximum of 1 year (31536000 seconds or 52560 blocks) is enforced as a sanity check.
+
+
+#### `CASTVOTE`
+```CASTVOTE <VOTENAME> <VOTE> <OPTION>```
+ - `VOTENAME`: should match a `VOTENAME` that was initiated and has not expired it's `DEADLINE`.
+ - `OPTION`: one of the options from `INITVOTE`.
+ - `VOTE`: int between 0 and 100 indicating percentage of vote.
+
+The tables where the poll data is stored and any validation errors of poll broadcast messages are excluded from consensus hashes,
+to make sure this is optional and can be disabled.
+
+#### Results
+Results will be `VOTE` * stake at the `STAKEHEIGHT` or otherwise the height at which `INITVOTE` is broadcasted.
+This can be on-the-fly queried from the database when fetching results.
+
+#### `LOCK`ed broadcast feed
+Counterparty has a feature to broadcast the text `LOCK` to prevent any futher broadcasts from that address ever being considered valid.
+These broadcasts are still parsed and stored, so it's possible to parse them for votes too if desired, but for now they are not!
```

Comment:
what does this mean? i.e. that voting with a locked feed works or doesn't work. the first part seems to state it will work fine, but the text "but for now they are not!" seems to negate that. please clarify